src/vfio_device.rs

```diff
@@ -616,7 +616,7 @@ impl VfioDeviceInfo {
 /// read/write/mmap offsets on the device descriptor, as well as mechanisms for describing and
 /// registering interrupt notifications.
 pub struct VfioDevice {
-    device: File,
+    device: ManuallyDrop<File>,
```

Comment:
I'm a bit concerned that having to do such thing might highlight a design mistake. The interaction between groups and device should be the other way around so that we don't need to make this manual drop, don't you think?

Reply:
It's a little odd, which is caused by hiding the VFIO group object from API.
The linux VFIO API has through classes of objects: container, group and device. 
Current vfio-ioctls crate only exposes container and device to the client, so caused the odd behavior.